test/ui/text_displayer_unit.js

```diff
@@ -42,12 +44,20 @@ describe('UITextDisplayer', () => {
   beforeAll(() => {
     videoContainer =
       /** @type {!HTMLElement} */ (document.createElement('div'));
+    videoContainer.style.height = `${videoContainerHeight}px`;
     document.body.appendChild(videoContainer);
     video = shaka.test.UiUtils.createVideoElement();
     videoContainer.appendChild(video);
+  });
+
+  beforeEach(() => {
     textDisplayer = new shaka.ui.TextDisplayer(video, videoContainer);
   });
 
+  afterEach(() => {
+    textDisplayer.destroy();
```

Comment:
destroy() is async, so please make this callback async and await the results of destroy().

Reply:
That implementation is not async today, but the underlying interfaces `shaka.extern.TextDisplayer` and `shaka.util.IDestroyable` are async and return a Promise.  In our codebase, any `destroy()` is async, and any `release()` is immediate.

It would be possible for us to add async steps to `destroy()` in `shaka.ui.TextDisplayer` at any time, and the tests should honor the contract of that API and use await.

Reply:
thanks for such a detailed explanation, now I've got it

done